x/ibc/23-commitment/merkle.go

```diff
@@ -93,11 +90,11 @@ func (Path) GetCommitmentType() Type {
 
 // String implements fmt.Stringer. It returns unescaped path of the URL string.
 func (p Path) String() string {
-	path, err := url.PathUnescape(p.KeyPath.String())
-	if err != nil {
-		panic(err)
-	}
-	return path
+	//path, err := url.PathUnescape(p.KeyPath.String())
```

Comment:
I see a new test failing here: 

```
--- FAIL: TestApplyPrefix (0.00s)
    merkle_test.go:115: 
        	Error Trace:	merkle_test.go:115
        	Error:      	Not equal: 
        	            	expected: "/storePrefixKey/path1/path2/path3/key"
        	            	actual  : "/storePrefixKey/path1%2Fpath2%2Fpath3%2Fkey"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/storePrefixKey/path1/path2/path3/key
        	            	+/storePrefixKey/path1%2Fpath2%2Fpath3%2Fkey
        	Test:       	TestApplyPrefix
        	Messages:   	Prefixed path incorrect
```

the problem is that `p.KeyPath.String()` does not represent the real path. That's why I added the `url.PathUnescape` method. 

cc: @AdityaSripal

Reply:
oh I see, so the test above is actually wrong.

Reply:
`tendermint/merkel` use the `/` as separator to split the multiple keys, `p.KeyPath.String()`  convert the key can avoid  the confusion.

Reply:
I believe the original method was correct. Keypath should be separated by `/`. Not sure what you mean here by multiple keys

Reply:
The `Keypath` is an array of Key which I mentioned multiple keys.
```
type Key struct {
	name []byte
	enc  keyEncoding
}

type KeyPath []Key
```
But the `Key.name` may contain `/`, so `KeyPath.String()` method escape the `Key.name`.
In this test case, there are two Keys with name `storePrefixKey` and `path1/path2/path3/key`. If we not escape the `Key.name`, we can't distinguish the `Key` and `Key.name`.